pieforproviders/pieforproviders
# [178] Pundit for attendances/billable occurrences, refactoring

# 💅🏼 What issue does this fix?


Finishes #178 with the new data model; we want Attendances to be created as instances of the polymorphic BillableOccurrence

I beefed up the request spec for displaying items in a user's scope; I create several records, only some of which belong to the current user, and make sure the number of displayed results is accurate.  I also refactored the requests specs because they were redefining several let blocks multiple times.

I changed all references to `item` in the request specs to `record` just because it has a little more meaning; I might even be convinced to call them `model` but I think that will cause confusion between requests specs and model specs.  I also added a shared example for updating nested items because of the new association between Attendances and BillableOccurrences.

## 🍂 Before You Submit


## 🧳 Steps to test

- run specs

- create several `BillableOccurrences` of type `Attendance` via the API (see the request spec for some ideas, and also the seeds file, where we make attendances by making `BillableOccurrences`) for several different users - the occurrences will need to be associated to some ChildApprovals, so make sure you create those required records first
- hit `/api/v1/billable_occurrences` and make sure:
	- you can only see the attendances for the current user
	- it includes the attendance data as well as the billable occurrence associated IDs

- create several business records as several different users
- hit `/api/v1/businesses` and make sure you can only see the current user's businesses
